packages/metadataform/src/metadataform.ts

```diff
@@ -416,12 +419,15 @@ export class MetadataFormWidget
       }
 
       // Fill the formData with the current metadata value.
-      if (hasValue) formData[metadataKey] = workingObject as PartialJSONValue;
+      if (hasValue) formData[metadataKey] = workingObject as JSONValue;
     }
 
     this.buildWidget({
       properties: formProperties,
-      settings: new Private.Settings(this._metaInformation),
+      settings: new Private.Settings({
+        metaInformation: this._metaInformation,
+        schema: this._metadataSchema
+      }),
```

Comment:
```suggestion
      settings: new BaseSettings({
        schema: this._metadataSchema as PartialJSONObject
      }),
```
Perhaps we can do without the `Settings` class, it seems to be used only to provide the functionalities of `BaseSettings`.
If so, the interface `ISettings` should be removed too.